numpy/lib/arraysetops.py

```diff
@@ -324,7 +325,15 @@ def _unique1d(ar, return_index=False, return_inverse=False,
         aux = ar
     mask = np.empty(aux.shape, dtype=np.bool_)
     mask[:1] = True
-    mask[1:] = aux[1:] != aux[:-1]
+    if aux.shape[0] > 0 and aux.dtype.kind in "cfmM" and np.isnan(aux[-1]):
+        # Ensure that `NaT` is used for time-like dtypes
+        nan = np.full(shape=1, fill_value=np.nan, dtype=aux.dtype)[0]
```

Comment:
I know that nan = np.full(shape=1, fill_value=np.nan, dtype=aux.dtype)[0] does not shine, but it does the work and the suggested np.array(np.nan, dtype=aux.dtype) was failing on ValueError: Could not convert object to NumPy datetime.

Reply:
Ah, annoying that `np.array(np.nan, dtype=aux.dtype)` doesn't work.
I'd argue that `np.array(np.nan).astype(aux.dtype)` would be a bit neater, but at this point things start to get nit picky.

Reply:
You are right. The `searchsorted` works with vector argument nicely.